src/Resources/Custom TypeEditor Template.md

```diff
@@ -0,0 +1,314 @@
+Custom TypeEditor Template
+
+Since .NET 3.1 started to support the WinForms Runtime, a new WinForms designer
+was needed to support .NET applications. The work required a near-complete
+rearchitecting of the designer, as we responded to the differences between .NET
+and the .NET Framework based WinForms designer everyone knows and loves.
+
+Until we added support for .NET Core applications there was only a single
+process, devenv.exe, that both the Visual Studio environment and the application
+being designed ran within. But .NET Framework and .NET Core can’t both run
+together within devenv.exe, and as a result we had to take the designer out of
+process, thus we called the new designer – WinForms Out of Process Designer (or
+OOP designer for short). We call the existing process – the process that Visual
+Studio runs in – the client process, and the process which shows the actual Form
+at Design time, the server process, which we call *DesignToolsServer*.
+
+# Enter the DesignToolsServer
+
+While simple Control Designer scenarios like Type Converters, Action Lists or
+CodeDom Serializers don’t need any substantial rewrites, Type Editors are a
+different beast altogether.
+
+To illustrate the problem created by introducing this DesignToolsServer as a
+dedicated server process, let’s look at a typical type editor for any image
+processing control property like the Button’s *BackgroundImage* property: While
+the actual Image that you picked will be rendered on a Button in the Server
+process, the dialog you picked it *with* runs in the context of Visual Studio.
```

Comment:
Type (where we refer to an _actual_ type) and properrties should formatted with backticks.


```suggestion
To illustrate the problem created by introducing this DesignToolsServer as a
dedicated server process, let’s look at a type editor for an image property like (e.g., the button’s `BackgroundImage` property).
While the actual image that you pick will be rendered in the server
process, the dialog you picked it *with* runs in the context of Visual Studio (i.e., the client process).
```